### ISSUE TITLE
Output result to terminal AND in special format

### ISSUE THREAD

**Issue by alexgit2k** [hadolint/hadolint#837]
- [] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

When integrating Hadolint in Gitlab with the codeclimate-report (see https://github.com/hadolint/hadolint/blob/master/docs/INTEGRATION.md#gitlab-ci) the result is only displayed if
1. a merge-request for the branch exists
**and**
2. the target-branch (of the merge-request) has already a codeclimate-report

If these conditions are not met and the check fails you will not see the errors in the pipeline-output (only in the artefact which needs to be downloaded and is in JSON-Format).

Therefore it would be good, if there are additional options for output-file/format, e.g.:
hadolint --ignore ... Dockerfile **--file reports/hadolint-$CI_COMMIT_SHA.json --file-format gitlab_codeclimate**

### Actual behavior

Output is redirected to a file
```hadolint Dockerfile --no-fail --format gitlab_codeclimate > reports/hadolint-$CI_COMMIT_SHA.json```

### Steps to reproduce the behavior

See https://github.com/hadolint/hadolint/blob/master/docs/INTEGRATION.md#gitlab-ci

### Workaround

Run hadlint twice:
```
  script:
    - mkdir -p reports
    - hadolint --ignore ... Dockerfile --no-fail --format gitlab_codeclimate > reports/hadolint-$CI_COMMIT_SHA.json
    - hadolint --ignore ... Dockerfile
```

**Comment by xlr-8** [comment]
You could also use `tee` to output it into several destinations without needing to re-run hadolint. That implies that `tee` binary is installed on the server/docker though, now that would also output JSON into your terminal - which isn't ideal either so, you could also use `jq .`

```bash
>  echo '{"foo": "bar"}' | tee -a file.log
{"foo": "bar"}
> cat file.log
{"foo": "bar"}
> jq . file.log
{
  "foo": "bar"
}
```

I know it doesn't quite solve your issue, but figured you might eventually save a bit of time instead of running it several times; and those might be of interested for those who don't know them.

**Comment by alexgit2k** [comment]
Interesting idea!

The two hadolint-runs only take 6 seconds including docker-start. So I think it's a lot of more work to find out the right parameters for jq to get near the default format output of hadolint.

**Comment by nejch** [comment]
In case maintainers don't like the idea of multiple outputs in a single run, I'm just linking an alternative approach that trivy uses now, which is a little `convert` subcommand that can achieve the same without expensive repeated calls.

https://aquasecurity.github.io/trivy/v0.52/docs/configuration/reporting/#converting

Since the example above uses GitLab, this is what our current trivy setup looks like using convert to save/show multiple different formats.

```yaml
lint-repo:
  script:
    - trivy fs --no-progress --ignore-unfixed --format json --output trivy-results.json $CI_PROJECT_DIR
    - trivy convert --format template --template "@/contrib/junit.tpl" --output trivy-junit.xml trivy-results.json
    - trivy convert --exit-code 1 --format table trivy-results.json
```

**Comment by JoshuaBehrens** [comment]
As I have php in use anyway I just wrote myself a little php script to convert hadolint json to junit

```shell
#!/usr/bin/env php
<?php

$hadolintResponse = \json_decode(\stream_get_contents(\STDIN), true);
$dockerFile = \array_pop($argv);

$countInfo = \count(\array_keys(\array_column($hadolintResponse, 'level'), 'info'));  // sync this threshold with .hadolint.yaml
$count = \count($hadolintResponse);

$xw = new XMLWriter();
$xw->openMemory();
$xw->startDocument("1.0");
    $xw->startElement("testsuites");
        $xw->startAttribute("failures");
            $xw->text($count - $countInfo);
        $xw->endAttribute();

        $xw->startAttribute("tests");
            $xw->text($count);
        $xw->endAttribute();

        $xw->startElement("testsuite");
            $xw->startAttribute("name");
                $xw->text("Project hadolint");
            $xw->endAttribute();

            $xw->startAttribute("file");
                $xw->text($dockerFile);
            $xw->endAttribute();

            foreach ($hadolintResponse as $hadolintLine) {
                $message = $hadolintLine['message'];
                $code = $hadolintLine['code'];
                $line = $hadolintLine['line'];
                $level = $hadolintLine['level'];
                
                $xw->startElement("testcase");
                    $xw->startAttribute("name");
                        $xw->text($code);
                    $xw->endAttribute();
    
                    $xw->startAttribute("line");
                        $xw->text($line);
                    $xw->endAttribute();
    
                    if ($level === 'info') { // sync this threshold with .hadolint.yaml
                        $xw->startElement("system-out");
                            $xw->text($message);
                        $xw->endElement();
                    } else {
                        $xw->startElement("failure");
                            $xw->startAttribute("message");
                                $xw->text($message);
                            $xw->endAttribute();
                        $xw->endElement();
                    }
                $xw->endElement();
            }
        $xw->endElement();
    $xw->endElement();
$xw->endDocument();

echo $xw->outputMemory();
```

**Comment by bcsgh** [comment]
Has anyone given a reason this shouldn't be done? Or is it just a matter of not having gotten around to it?

**Comment by amai2012** [comment]
Can this issue be closed since https://github.com/hadolint/hadolint/pull/1197 ß

**Comment by m-ildefons** [comment]
Yes, I think so.
